server/datastore/mysql/microsoft_mdm_test.go

```diff
@@ -479,6 +479,54 @@ func testMDMWindowsDiskEncryption(t *testing.T, ds *Datastore) {
 			cleanupHostProfiles(t)
 		})
 
+		t.Run("BitLocker profile status with PIN required", func(t *testing.T) {
```

Comment:
These tests are very inter-related.  I tried to make them more table-based (I really did, @ksykulev!) but somehow starting from scratch caused unexpected issues as well, perhaps due to mdm side-effects I don't fully grok.  The best I could do is create a mini setup and teardown in this test -- it turns on PIN requirement at the beginning, and removes it at the end.  It also reverts the change made to the host before returning.

Reply:
I think changing existing tests is going to be fairly arduous. Especially in some of the extra complicated situations. I have begun pulling out any new tests into new methods in the new paradigm - as long as the setup isn't wildly complicated.